admin/client/src/state/formData/FormDataReducer.js

```diff
@@ -0,0 +1,43 @@
+import deepFreeze from 'deep-freeze-strict';
+import ACTION_TYPES from './FormDataActionTypes';
+
+const initialState = deepFreeze({});
+
+export default function schemaReducer(state = initialState, action = null) {
+  switch (action.type) {
+    case ACTION_TYPES.SET_SCHEMA_STATE_OVERRIDES: {
+      return deepFreeze(Object.assign({}, state, {
+        [action.payload.id]: Object.assign({}, state[action.payload.id], {
+          stateOverride: action.payload.stateOverride,
+        }),
+      }));
+    }
+
+    case ACTION_TYPES.DESTROY_SCHEMA_STATE_OVERRIDES: {
+      return deepFreeze(Object.assign({}, state, {
+        [action.payload.id]: Object.assign({}, state[action.payload.id], {
+          stateOverride: undefined,
+        }),
+      }));
+    }
+
+    case ACTION_TYPES.SET_SCHEMA_LOADING: {
+      return deepFreeze(Object.assign({}, state, {
+        [action.payload.id]: Object.assign({}, state[action.payload.id], {
+          loading: true,
+        }),
+      }));
+    }
+
+    case ACTION_TYPES.SET_SCHEMA_LOADED: {
```

Comment:
Distinguishing between `SET_SCHEMA_LOADING` and `SET_SCHEMA_LOADED` seems a bit clunky, easy to misread - could you just pass through a boolean on `SET_SCHEMA_LOADING`?